Metaverse/Source/Metaverse/MetaverseCharacter.cpp

```diff
@@ -153,11 +153,14 @@ void AMetaverseCharacter::InitializeMetaverseOverlay() {
         Cast<APlayerController>(GetController());
     AMetaverseHUD *MetaverseHUD =
         Cast<AMetaverseHUD>(PlayerController->GetHUD());
-    MetaverseOverlay = MetaverseHUD->GetMetaverseOverlay();
-    MetaverseOverlay->SetBalance(0);
-    MetaverseOverlay->SetCoin(0);
-    MetaverseOverlay->SetAccount(
-        FText::FromString(FString::Printf(TEXT("Please connect a WalletConnect-compatible wallet"))));
+    // if MetaverseHUD is valid
+    if (MetaverseHUD) {
```

Comment:
I guess this is a raw pointer, so this check should be enough?

Reply:
yes, checking pointer is enough, actually we need to check every pointer before accessing them...

Reply:
but the code would become very ugly

Reply:
But here, I guess, checking MetaverseHUD null is redundant, since GetHUD should always return non-null value (because in game mode, it is already set a valid HUD).

But no harm to check pointer..

Reply:
If it is null, we just avoid the possible crashing problem instead of fixing it: https://github.com/cronos-labs/play-unreal-plugin/issues/28